kytos/core/exceptions.py

```diff
@@ -77,16 +77,70 @@ def __str__(self):
         return msg
 
 
-class KytosSetTagRangeError(Exception):
+class KytosLinkCreationError(Exception):
+    """Exception thrown when the link has an empty endpoint."""
+
+
+class KytosTagError(Exception):
+    """Exception to catch an error when setting tag type or value."""
+    def __init__(self, msg: str) -> None:
+        self.msg = msg
+
+    def __str__(self) -> str:
+        return f"{self.msg}"
+
+    def __repr__(self) -> str:
+        return f"{self.msg}"
+
+
+class KytosTagtypeNotSupported(KytosTagError):
+    """Exception thrown when a not supported tag type is not supported"""
+    def __init__(self, msg: str) -> None:
+        super().__init__(f"KytosTagtypeNotSupported, {msg}")
+
+
+class KytosInvalidTagRanges(KytosTagError):
+    """Exception thrown when a list of ranges is invalid."""
+    def __init__(self, msg: str) -> None:
+        super().__init__(f"KytosInvalidTagRanges, {msg}")
+
+
+class KytosSetTagRangeError(KytosTagError):
     """Exception raised when available_tag cannot be resized"""
+    def __init__(self, msg: str) -> None:
+        super().__init__(f"KytosSetTagRangeError, {msg}")
 
 
-class KytosLinkCreationError(Exception):
-    """Exception thrown when the link has an empty endpoint."""
+class KytosTagsNotInTagRanges(Exception):
+    """Exception thrown when a tag is outside of tag ranges"""
+    def __init__(self, conflict: list[list[int]], intf_id: str) -> None:
+        super().__init__()
+        self.conflict = conflict
+        self.intf_id = intf_id
+
+    def __repr__(self):
+        return f"The tags {self.conflict} are outside tag_ranges"\
+               f" in {self.intf_id}"
+
+    def __str__(self) -> str:
+        return f"The tags {self.conflict} are outside tag_ranges"\
+               f" in {self.intf_id}"
+
+
+class KytosTagsAreNotAvailable(Exception):
+    """Exception thrown when a tag is not available."""
+    def __init__(self, conflict: list[list[int]], intf_id: str) -> None:
+        super().__init__()
+        self.conflict = conflict
+        self.intf_id = intf_id
 
+    def __repr__(self):
+        return f"The tags {self.conflict} are not available."\
```

Comment:
Trivial detail but the `"."` ended up misplaced as far as I can tell:

```
2023-11-09 12:45:53,743 - ERROR [kytos.napps.kytos/of_lldp] [main.py:301:use_vlan] (thread_pool_app_2) The tags [[5799, 5799]] are not available. in 00:00:00:00:00:00:00:01:5
```

Also, if since you've implemented `__repr__` you can omit `__str__`, so feel free to nuke all `__str__` when `__repr__` has also been covered.

Reply:
Fixed. Also, the new exceptions were added under `KytosTagError`